### ISSUE TITLE
oops! problem with continuity

### ISSUE THREAD

**Issue by jonsterling** [jonsterling/nominal-lcf-paper#5]
I just realized one problem with the "denotational semantics" approach that I'm taking, which is that whilst we can calculate the modulus of continuity, we cannot do so extensionally/continuously. So, the resulting interpretations aren't really functions like I claimed they were...

I wonder what can be done.

**Comment by jonsterling** [comment]
Hmm, this may not actually be a problem. Here's the place where we use it (look for `MOD`):
![screen shot 2016-04-15 at 3 09 13 pm](https://cloud.githubusercontent.com/assets/55057/14576203/106bd304-031c-11e6-8ce7-e15113b0e303.png)

At the point that we are building `M'`, the "discontinuous" part (the calculation of the modulus of continuity from the functional `M`) has already happened. The calculation happened at the meta-level, so there is no need to worry about that; the `MOD` functional is not actually part of the interpretation of a term, only the result of *applying* the `MOD` functional to something.

So, I think there's no problem at all here. **But I'll have to make a note of it in the paper, since probably others will have a similar complaint.**

**Comment by jonsterling** [comment]
Well, I just realized that this actually still kind of suspect, but it should be possible to fix it! The problem is currently that the call to `Mod` depends on the input to `M'`, so I don't think we can actually claim that `M'` is continuous as written.

So, suppose we've got a functional `F : D^ω -> E`; then, from continuity we can exhibit a new functional `Mod(F) : D^ω -> Nat`, which is the modulus of continuity of `F`. Now, `Mod` is not a continuous function, but `Mod(F)` should be (let's double check this).

If we view tactics in uncurried form like `M : Atom^ω * State(Jdg) -> State(Jdg)`, then we can get the modulus of continuity all at once, `Mod(M) : Atom^ω * State(Jdg) -> Nat`. Then, assuming my claim above about the continuity of the modulus of continuity once you've got it is right, THEN we should be able write out the interpretation in a way that's "clearly" continuous.
